### PR TITLE
Add lsp-javascript backend for javascript layer

### PR DESCRIPTION
**Pull request by ztlevi**
Added `lsp-javascript` for javascript layer. Tested `company`, `spacemacs-jump-handlers-js2-mode` for both tern and lsp.

### CODE REVIEW

File: layers/+lang/javascript/README.org [hunk 1]
```diff
@@ -85,6 +107,14 @@ layer:
   (javascript :variables tern-command '("node" "/path/to/tern/bin/tern"))
 #+END_SRC
 
+** Language Server Protocol
+You just have to install python language server package:
```
Comment (Miloas) [inline layers/+lang/javascript/README.org]: javascript and typescript language server here

**Comment by CestDiego** [comment]
This is amazing.

File: layers/+tools/lsp/funcs.el [hunk 2]
```diff
@@ -30,3 +30,8 @@
                       :background (face-attribute 'highlight :background nil t)
                       :foreground (face-attribute 'default :foreground nil t))
   )
+
+(defun spacemacs//set-lsp-key-bindings (mode)
+  "Set the key bindings for tern and the given MODE."
```
Comment (CestDiego) [inline layers/+tools/lsp/funcs.el]: keybindings for lsp

Comment (syl20bnr) [reply layers/+tools/lsp/funcs.el]: I fixed it, then I renamed the function as it does not set the key bindings but the jump handler, see commit:  https://github.com/syl20bnr/spacemacs/commit/7596a156a

**Comment by syl20bnr** [comment]
Thank you ! 💯 💜 

I renamed the function `spacemacs//set-lsp-key-bindings` as it does not set the key bindings but the jump handler, see commit:  https://github.com/syl20bnr/spacemacs/commit/7596a156a

Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**